lib/cn/kvset.c

```diff
@@ -3320,7 +3360,19 @@ kvset_iter_next_vref(
             break;
     }
 
+    if ((*vtype == vtype_val || *vtype == vtype_cval) && ks->ks_use_vgmap) {
+        merr_t err;
+
+        /* This ugly cast is because we use a mix of 32 and 16-bit bytes to represent
+         * vbidx in memory
+         */
+        err = vgmap_vbidx_src2out(ks->ks_vgmap, *vbidx, (uint16_t *)vbidx);
+        if (err)
+            abort();
```

Comment:
Is this the best course of action? Should we bubble the error instead?

Reply:
Best course of action until we update this function to return an error. Will be done in a follow-on PR.